saleor/core/utils/random_data.py

```diff
@@ -562,32 +568,256 @@ def create_fake_user(user_password, save=True, generate_id=False):
+    refund_psp = str(uuid.uuid4())
+
+    if flow == "authorization":
+        pick = random.random()
+        if pick < 0.15:
+            # Authorized only: actions -> CHARGE, CANCEL
+            return create_transaction_with_events(
+                order=order,
+                name="",
+                app=dummy_app,
+                app_identifier=dummy_app_identifier,
+                user=None,
+                available_actions=[TransactionAction.CHARGE, TransactionAction.CANCEL],
+                authorized_amount=amount,
+                charge_amount=None,
+                refund_amount=None,
+                authorization_psp=auth_psp,
+                charge_psp=None,
+                refund_psp=None,
+            )
+        if pick < 0.75:
+            # Authorized + Charged: actions -> REFUND
+            return create_transaction_with_events(
+                order=order,
+                name="",
+                app=dummy_app,
+                app_identifier=dummy_app_identifier,
+                user=None,
+                available_actions=[TransactionAction.REFUND],
+                authorized_amount=amount,
+                charge_amount=amount,
+                refund_amount=None,
+                authorization_psp=auth_psp,
+                charge_psp=charge_psp,
+                refund_psp=None,
+            )
+        # Authorized + Charged + Refunded
+        return create_transaction_with_events(
+            order=order,
+            name="",
+            app=dummy_app,
+            app_identifier=dummy_app_identifier,
+            user=None,
+            available_actions=[],
+            authorized_amount=amount,
+            charge_amount=amount,
+            refund_amount=amount,
+            authorization_psp=auth_psp,
+            charge_psp=charge_psp,
+            refund_psp=refund_psp,
+        )
+    # Charge strategy:
+    if random.random() < 0.8:
+        # Charged only
+        return create_transaction_with_events(
+            order=order,
+            name="",
+            app=dummy_app,
+            app_identifier=dummy_app_identifier,
+            user=None,
+            available_actions=[TransactionAction.REFUND],
+            authorized_amount=None,
+            charge_amount=amount,
+            refund_amount=None,
+            authorization_psp=None,
+            charge_psp=charge_psp,
+            refund_psp=None,
+        )
+    # Charged + Refunded
+    return create_transaction_with_events(
         order=order,
-        payment_token=str(uuid.uuid4()),
-        total=order.total.gross.amount,
-        currency=order.total.gross.currency,
+        name="",
+        app=dummy_app,
+        app_identifier=dummy_app_identifier,
+        user=None,
+        available_actions=[],
+        authorized_amount=None,
+        charge_amount=amount,
+        refund_amount=amount,
+        authorization_psp=None,
+        charge_psp=charge_psp,
+        refund_psp=refund_psp,
     )
```

Comment:
You can simplify it a little. Define `kwargs` at the beginning and only update specific fields, and then use `**kwargs` in function call. Like below:
```suggestion
    kwargs = {
        "order": order,
        "name": "",
        "app": dummy_app,
        "app_identifier": dummy_app_identifier,
        "user": None,
        "available_actions": [],
        "authorized_amount": None,
        "charge_amount": None,
        "refund_amount": None,
        "authorization_psp": None,
        "charge_psp": None,
        "refund_psp": None,
    }
    if flow == "authorization":
        pick = random.random()
        if pick < 0.15:
            kwargs["actions"] = [TransactionAction.CHARGE, TransactionAction.CANCEL]
            kwargs["authorized_amount"] = amount
            kwargs["authorization_psp"] = auth_psp
            return create_transaction_with_events(**kwargs)
    ...
```